arithmetic_analysis/newton_method.py

```diff
@@ -36,7 +36,7 @@ def newton(
         try:
             next_guess = prev_guess - function(prev_guess) / derivative(prev_guess)
         except ZeroDivisionError:
```

Comment:
What would be the downside of:
```suggestion
        except ZeroDivisionError as e:
            raise ZeroDivisionError("Could not find root") from e
```

Reply:
This will still give you two `traceback` objects:
```python
Traceback (most recent call last):
  File "arithmetic_analysis/newton_method.py", line 37, in newton
    next_guess = prev_guess - function(prev_guess) / derivative(prev_guess)
ZeroDivisionError: float division by zero

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "arithmetic_analysis/newton_method.py", line 54, in <module>
    print(newton(f, f1, 3))
  File "arithmetic_analysis/newton_method.py", line 39, in newton
    raise ZeroDivisionError("Could not find root") from err
ZeroDivisionError: Could not find root
```

`raise` inside an `except` block is mostly used to change the exception to output to the user.

I also don't understand if it is meant to be raising an error then why not let it raise itself? Why use `raise` inside an `except` block? `except` block is used to **handle** the errors.